File: src/main/scala/chapter1/Student.scala
package chapter1

class Student(name: String, age: Int) {
  def printInfo(): Unit = {
    println(this.name + "\t" + this.age + "\t" + Student.school)
  }
}

//引入伴生对象
object Student {
  val school: String = "atguigu"

  def main(args: Array[String]): Unit = {
    val alice = new Student(name = "alice", age = 18)
    val bob = new Student(name = "bob", age = 32)
    alice.printInfo()
    bob.printInfo()

  }
}

File: src/main/java/Student.java
public class Student {
    private String name;
    private Integer age;
    private static String school = "atguigu";


    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }

    public void printInfo() {
        System.out.println(this.name + "\t" + this.age + "\t" + Student.school);
    }

    public static void main(String[] args) {
        Student alice = new Student("alice", 32);
        Student bob = new Student("bob", 32);
        alice.printInfo();
        bob.printInfo();
    }
}
